### ISSUE TITLE
Can't connect to API when URL contains encoded slash

### ISSUE THREAD

**Issue by sergejmueller** [grafana/grafana-json-datasource#59]
Hello.

Whenever I want to create a data source with the JSON API plugin I get an error message: “JSON API: Cannot connect to API”. But only for URLs that have an encoded slash, e.g. https://streamr.network/api/v1/streams/0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fethereum-gas-prices/data/partitions/0/last - I can request the URL in a browser without errors.

Any idea? Should I encode/escape any character more?
Thank you.

**Comment by marcusolsson** [comment]
Thank you for letting me know, and for providing an endpoint that I can easily test.

I confirm that I'm able to reproduce this. Indeed it looks like the encoded slash gets decoded. 

<img width="826" alt="CleanShot 2021-02-14 at 11 22 55@2x" src="https://user-images.githubusercontent.com/8396880/107874032-05c81000-6eb7-11eb-8e82-5782874a0364.png">

![CleanShot 2021-02-14 at 11 23 41@2x](https://user-images.githubusercontent.com/8396880/107874064-2bedb000-6eb7-11eb-8603-44009ea1db60.png)

I don't have a quick fix in mind at this point, but I'll make sure to look into this.

**Comment by sergejmueller** [comment]
@marcusolsson Thank you for fixing the issue. How I get the updated plugin version on grafana.com (Grafana Cloud)?

**Comment by marcusolsson** [comment]
The fix is available in v1.0.1. You should be able to update it by installing the latest version.

**Comment by sergejmueller** [comment]
I know, I can update any plugin on my local Grafana instance. The question is, how I can update a plugin on the Grafana Cloud?
<img width="1680" alt="Bildschirmfoto 2021-03-06 um 13 44 41" src="https://user-images.githubusercontent.com/108992/110207195-2108b880-7e82-11eb-8363-4f3c85be6649.png">

**Comment by marcusolsson** [comment]
You should be able to upgrade from the marketplace. If you go to this URL:

https://grafana.com/grafana/plugins/marcusolsson-json-datasource/installation?version=1.0.1

Do you see this button?

<img width="158" alt="Screenshot 2021-03-06 at 13 56 49@2x" src="https://user-images.githubusercontent.com/8396880/110207448-d38d4b00-7e83-11eb-90ac-6610a2bcee64.png">

**Comment by sergejmueller** [comment]
Ah, ok. I see. Updated. (not a user friendly way to updated a plugin 😀)

**Comment by sergejmueller** [comment]
Marcus, did you test your fix with the url above `https://streamr.network/api/v1/streams/0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fethereum-gas-prices/data/partitions/0/last`?

After the update i get an error "Stream 0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fstreamr%2Fgrowth%2Fstreams not found."

<img width="1680" alt="Bildschirmfoto 2021-03-06 um 14 09 45" src="https://user-images.githubusercontent.com/108992/110207711-9e81f800-7e85-11eb-86bb-21f2283ba7ef.png">

**Comment by marcusolsson** [comment]
It looks like it's working as intended, no? Are you sure the resource exists?

**Comment by sergejmueller** [comment]
Of course, the resource is available and returns a valid JSON https://streamr.network/api/v1/streams/0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fethereum-gas-prices/data/partitions/0/last

**Comment by marcusolsson** [comment]
Hm. Weird thing is that you're getting an API error saying that the resource doesn't exist. And the resource name looks correct? Not sure what's happening here.

**Comment by sergejmueller** [comment]
The error message says, the stream is not found. The stream is the encoded part of the url: `0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fethereum-gas-prices`. If you click the link https://streamr.network/api/v1/streams/0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fethereum-gas-prices/data/partitions/0/last you can see, the browser can resolve the encoded part and the API endpoint returns a valid JSON.

The Grafana JSON plugin (or Grafana Core?) creates a broken request url, so that the API endpoint can't resolve the encoded stream within the requested url.

**Let me explain**
Original URL: `https://streamr.network/api/v1/streams/0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fethereum-gas-prices/data/partitions/0/last`
Builded / Requested URL (from Grafana Query Inspector): `api/datasources/proxy/21/api/v1/streams/0x688759bcbb6adf32a07e91f6de84d181b252e655%252Fethereum-gas-prices/data/partitions/0/last`

Please compare the encoded slashes:
Original URL: `%2F` (correct)
Builded / Requested URL: `%252F` (wrong because double encoded)

<img width="1680" alt="Bildschirmfoto 2021-03-07 um 08 58 45" src="https://user-images.githubusercontent.com/108992/110233390-08e97580-7f24-11eb-88ea-56fe89676864.png">

**Comment by marcusolsson** [comment]
This is weird. It seems like there's some magic happening in the Grafana proxy. 

If I make the request through the proxy, it fails. 

```
http://localhost:3000/api/datasources/proxy/51/api/v1/streams/0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fethereum-gas-prices/data/partitions/0/last
```

But if I make the request directly from the browser, it succeeds even without the earlier "fix". 

```
https://streamr.network/api/v1/streams/0x688759bcbb6adf32a07e91f6de84d181b252e655%2Fethereum-gas-prices/data/partitions/0/last
```

I'll have to look closer at what the proxy does with the request. I've reverted the previous commit for now. Thanks for the patience on this.

**Comment by jskiweb** [comment]
Hi, i know its now 2022, But i did try to setup an json api  (http://IP/api/v2/status) and it is not working it says network error, But on the raspberry pi where my grafana is installed , a normal curl request works fine. I did try to setup another api without a slash and it works. Any fix for that?
